frontend/src/createStream/StreamingTable.js

```diff
@@ -1,109 +1,123 @@
+import { Grid } from "@material-ui/core";
 
 import axios from "axios";
 
-
 import SelectDevicesTable from "./SelectDevicesTable";
 import StreamButton from "../General/Buttons/StreamButton";
 
 export default class StreamingTable extends React.Component {
-
-    constructor(props) {
-        super(props)
-        this.state = {
-            senders: [],
-            receivers: [],
-            selectedSenderID: "",
-            selectedReceiverID: ""
-        }
-
-        this.dataSource = props.dataSource;
-        this.handleSendersChange = this.handleSendersChange.bind(this);
-        this.handleReceiversChange = this.handleReceiversChange.bind(this);
-
-        this.onSenderSelected = this.onSenderSelected.bind(this);
-        this.onReceiverSelected = this.onReceiverSelected.bind(this);
-
-        this.handleSubmit = this.handleSubmit.bind(this);
-    }
-
-    componentDidMount() {
-        this.dataSource.getSenders(this.handleSendersChange);
-        this.dataSource.getReceivers(this.handleReceiversChange);
-    }
-
-    handleSendersChange(senders) {
-        this.setState({
-            senders: senders
-        });
-    }
-
-    handleReceiversChange(receivers) {
-        this.setState({
-            receivers: receivers
-        });
-    }
-
-    onSenderSelected(selectedSender) {
-
-        this.setState({
-            selectedSenderID: selectedSender.target.value
-        })
-    }
-
-    onReceiverSelected(selectedReceiver) {
-
-        this.setState({
-            selectedReceiverID: selectedReceiver.target.value
+  constructor(props) {
+    super(props);
+    this.state = {
+      senders: [],
+      receivers: [],
+      selectedSenderID: "",
+      selectedReceiverID: "",
+    };
+
+    this.dataSource = props.dataSource;
+    this.handleSendersChange = this.handleSendersChange.bind(this);
+    this.handleReceiversChange = this.handleReceiversChange.bind(this);
+
+    this.onSenderSelected = this.onSenderSelected.bind(this);
+    this.onReceiverSelected = this.onReceiverSelected.bind(this);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.dataSource.getSenders(this.handleSendersChange);
+    this.dataSource.getReceivers(this.handleReceiversChange);
+  }
+
+  handleSendersChange(senders) {
+    this.setState({
+      senders,
+    });
+  }
+
+  handleReceiversChange(receivers) {
+    this.setState({
+      receivers,
+    });
+  }
+
+  onSenderSelected(selectedSender) {
+    this.setState({
+      selectedSenderID: selectedSender.target.value,
+    });
+  }
+
+  onReceiverSelected(selectedReceiver) {
+    this.setState({
+      selectedReceiverID: selectedReceiver.target.value,
+    });
+  }
+
+  handleSubmit(event) {
+    if (
+      this.state.selectedReceiverID === "" ||
+      this.state.selectedSenderID === ""
+    ) {
+      console.log("Please select a sender and a receiver");
+    } else {
+      axios
+        .post("http://localhost:8080/stream", {
```

Comment:
How can we disable this particular formatting rule. This is so unclear.

Reply:
This particular formatting rule has to do with [require a newline after each call in a method chain](https://eslint.org/docs/rules/newline-per-chained-call), and possibly the enforcement of a maximum line length. The formatting is pretty much set to do this unless we allow for really long lines of code. I could not find an option to ignore only the first call in a chain, so, I think it would be best to stick with the default here.